### PR TITLE
Support doubles in v8::Number

### PR DESCRIPTION
**Pull request by 190n**
### What does this PR do?

Supports all double-precision numbers in v8::Number, rather than only 32-bit signed ints

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests

### CODE REVIEW

**Comment by robobun** [comment]
@190n, your commit 74afbb5640fe84c83395b2d1be28f47b0c728270 is building: [`#1940`](https://buildkite.com/bun/bun/builds/1940)

File: src/bun.js/bindings/v8/V8HandleScope.h [hunk 1]
```diff
@@ -24,6 +24,10 @@ class HandleScope {
             return Local<T>(buffer->createHandle(value.asCell(), &Map::object_map));
         } else if (value.isInt32()) {
             return Local<T>(buffer->createSmiHandle(value.asInt32()));
+        } else if (value.isNumber()) {
+            double numeric_value = value.asNumber();
+            void* double_reinterpreted_to_pointer = *reinterpret_cast<void**>(&numeric_value);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/v8/V8HandleScope.h:29]: lol

File: src/bun.js/bindings/v8/V8HandleScopeBuffer.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ void HandleScopeBuffer::visitChildrenImpl(JSCell* cell, Visitor& visitor)
     for (int i = 0; i < thisObject->size; i++) {
         auto& handle = thisObject->storage[i];
         if (handle.isCell()) {
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/v8/V8HandleScopeBuffer.cpp:32]: are these zero initialized?  or whats the default vlaue?

File: src/bun.js/bindings/v8/V8Data.h [hunk 3]
```diff
@@ -55,8 +55,12 @@ class Data {
                 return JSC::jsBoolean(false);
             }
 
-            JSC::JSCell** v8_object = reinterpret_cast<JSC::JSCell**>(raw_ptr);
-            return JSC::JSValue(v8_object[1]);
+            ObjectLayout* v8_object = reinterpret_cast<ObjectLayout*>(raw_ptr);
+            if (v8_object->map.getPtr<Map>()->instance_type == InstanceType::HeapNumber) {
+                return JSC::jsNumber(*reinterpret_cast<double*>(&v8_object->ptr));
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/v8/V8Data.h]:
```suggestion
                return JSC::jsDoubleNumber(*reinterpret_cast<double*>(&v8_object->ptr));
```

File: src/bun.js/bindings/v8/V8HandleScopeBuffer.cpp (hunk 2 above)
Comment (190n) [reply src/bun.js/bindings/v8/V8HandleScopeBuffer.cpp:32]: It's not super clear but `to_v8_object` and `map` get initialized to null pointers, while `ptr` is uninitialized. I'll improve the default `Handle` constructor on the same branch where I make the other changes to `HandleScopeBuffer` we were talking about.

**Pull request merged**